### PR TITLE
Tmedia-745:  large manual promo block refactor

### PR DESCRIPTION
**Pull request by malavikakoppula**
**If you have not filled out the checklist below, the pr is not ready for review.**

## Description

- Updated Large Manual promo block to use v2 architecture
- Updated test cases and README

## Jira Ticket

- [TMEDIA-745](https://arcpublishing.atlassian.net/browse/TMEDIA-745)

## Acceptance Criteria

No special notes in Jira ticket. Followed [checklist](https://arcpublishing.atlassian.net/wiki/spaces/TI/pages/3106243821/Converting+a+v1+block+to+v2+architecture) for migrating blocks from v1 to v2.

## Test Steps

- Checkout this branch git checkout TMEDIA-745-large-manual-promo-block-refactor
- Run storybook with npm run storybook and verify functionality corresponding with stories.

### Before

- Using V1 architecture

Screen Shot 2022-06-30 at 10 35 08 AM
Screen Shot 2022-06-30 at 10 35 24 AM

### After

- Migrated to V2 architecture
Screen Shot 2022-07-05 at 10 26 05 AM
Screen Shot 2022-07-05 at 10 26 23 AM

## Author Checklist

- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [x] Confirmed this PR has unit test files
  - [x] Ran `npm run test`, made sure all tests are passing
  - [x] If the amount of work to write unit tests for this change are excessive,
        please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

File: .storybook/themes/news.scss [hunk 1]
```diff
@@ -1368,6 +1384,25 @@
 						),
 					),
 				),
+				large-manual-promo: (
+					components: (
+						stack: (
+							gap: var(--global-spacing-2),
```
Comment (matthewroach) [inline .storybook/themes/news.scss]: If the value does not change between default and desktop we don't need to add the same value here

File: blocks/large-manual-promo-block/features/large-manual-promo/default.jsx [hunk 2]
```diff
@@ -50,6 +167,18 @@ LargeManualPromo.propTypes = {
 			defaultValue: false,
 			group: "Configure Content",
 		}),
+		author: PropTypes.string.tag({
```
Comment (matthewroach) [inline blocks/large-manual-promo-block/features/large-manual-promo/default.jsx]: We should not be adding additional content as part of this work - that is out of scope and not part of this deliverable

File: blocks/large-manual-promo-block/_index.scss [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+		@include scss.block-properties("large-manual-promo-info");
+		@include scss.block-components("large-manual-promo-info");
+	}
+
+	&__heading {
+		@include scss.block-properties("large-manual-promo-right-heading");
+		@include scss.block-components("large-manual-promo-right-heading");
+	}
+
+	&__badge {
```
Comment (matthewroach) [inline blocks/large-manual-promo-block/_index.scss]: I do not see this class name within the block HTML

File: blocks/large-manual-promo-block/features/large-manual-promo/default.jsx [hunk 4]
```diff
@@ -1,19 +1,136 @@
...
+					headline
+				)}
+			</Heading>
+		) : null;
+
+	const PromoOverline = () => {
+		if (showOverline) {
+			if (overlineURL) {
+				return (
+					<Overline href={overlineURL} className={`${BLOCK_CLASS_NAME}__overline`}>
```
Comment (matthewroach) [inline blocks/large-manual-promo-block/features/large-manual-promo/default.jsx]: We do not need the classname applied to the `Overline` component as you can style the component via the `components: ()` map of the block

File: blocks/large-manual-promo-block/features/large-manual-promo/default.jsx [hunk 5]
```diff
@@ -1,19 +1,136 @@
...
+				suppressContentEditableWarning
+			>
+				<Image alt={headline} src={imageURL} searchableField />
+			</MediaItem>
+		) : null;
+	};
+
+	const PromoHeading = () =>
+		showHeadline ? (
+			<Heading className={`${BLOCK_CLASS_NAME}__heading`}>
```
Comment (matthewroach) [inline blocks/large-manual-promo-block/features/large-manual-promo/default.jsx]: We do not need the classname applied to the `Heading` component as you can style the component via the `components: ()` map of the block

File: blocks/large-manual-promo-block/features/large-manual-promo/default.jsx [hunk 6]
```diff
@@ -1,19 +1,136 @@
...
+				);
+			}
+			return <Overline className={`${BLOCK_CLASS_NAME}__overline`}>{overline}</Overline>;
+		}
+		return null;
+	};
+
+	const PromoDescription = () =>
+		showDescription ? (
+			<Paragraph className={`${BLOCK_CLASS_NAME}__description`}>{description}</Paragraph>
```
Comment (matthewroach) [inline blocks/large-manual-promo-block/features/large-manual-promo/default.jsx]: We do not need the classname applied to the `Paragraph` component as you can style the component via the `components: ()` map of the block

File: blocks/large-manual-promo-block/features/large-manual-promo/default.jsx [hunk 7]
```diff
@@ -1,19 +1,136 @@
...
+		) : null;
+
+	const PromoAttribution = () => (
```
Comment (matthewroach) [inline blocks/large-manual-promo-block/features/large-manual-promo/default.jsx]: This is an addition - we should not be adding features

**Review by matthewroach — DISMISSED** [review]

**Comment by JackHowa** [comment]
@malavikakoppula is this ready for a re-review? I see that it's in a draft state and has wip in the title?

File: blocks/large-manual-promo-block/features/large-manual-promo/default.jsx [hunk 8]
```diff
@@ -50,6 +143,10 @@ LargeManualPromo.propTypes = {
 			defaultValue: false,
 			group: "Configure Content",
 		}),
+		dateString: PropTypes.string.tag({
```
Comment (matthewroach) [inline blocks/large-manual-promo-block/features/large-manual-promo/default.jsx]: Should be removed

File: blocks/large-manual-promo-block/features/large-manual-promo/default.jsx [hunk 9]
```diff
@@ -1,19 +1,112 @@
...
 	}
+
+	const PromoImage = () => {
+		const { searchableField } = useEditableContent();
+
+		return showImage ? (
+			<MediaItem
+				caption={!hideImageCaption ? caption : null}
+				credit={!hideImageCredits ? credits : null}
+				title={!hideImageTitle ? subtitle : null}
```
Comment (matthewroach) [inline blocks/large-manual-promo-block/features/large-manual-promo/default.jsx]: Caption, Credit and Title are additional functionality we can remove
```suggestion
```

File: blocks/large-manual-promo-block/features/large-manual-promo/default.jsx [hunk 10]
```diff
@@ -1,19 +1,112 @@
...
+	const PromoDescription = () => (showDescription ? <Paragraph>{description}</Paragraph> : null);
+
 	return (
-		<LazyLoad enabled={customFields?.lazyLoad && !isAdmin}>
-			<LargePromoPresentation {...customFields} />
+		<LazyLoad enabled={shouldLazyLoad}>
+			<HeadingSection>
+				<Grid
+					role="article"
+					className={`${BLOCK_CLASS_NAME} ${BLOCK_CLASS_NAME}__${imagePosition}`}
```
Comment (matthewroach) [inline blocks/large-manual-promo-block/features/large-manual-promo/default.jsx]: The image position class is a modifier we should use the BEM syntax hyphens `--` 
```suggestion
					className={`${BLOCK_CLASS_NAME} ${BLOCK_CLASS_NAME}--${imagePosition}`}
```

Comment (matthewroach) [reply blocks/large-manual-promo-block/features/large-manual-promo/default.jsx]: I don't see a CSS class in the Sass file for this. Is this needed?

**Comment by malavikakoppula** [comment]
> @malavikakoppula is this ready for a re-review? I see that it's in a draft state and has wip in the title?

Just push some more changes and updating the PR. Will takeout the WIP shortly. Thanks for checking

File: .storybook/themes/news.scss (hunk 1 above)
Comment (malavikakoppula) [reply .storybook/themes/news.scss]: Removed it.

File: blocks/large-manual-promo-block/features/large-manual-promo/default.jsx (hunk 2 above)
Comment (malavikakoppula) [reply blocks/large-manual-promo-block/features/large-manual-promo/default.jsx]: Removed it.

File: blocks/large-manual-promo-block/_index.scss (hunk 3 above)
Comment (malavikakoppula) [reply blocks/large-manual-promo-block/_index.scss]: Removed it.

File: blocks/large-manual-promo-block/features/large-manual-promo/default.jsx (hunk 4 above)
Comment (malavikakoppula) [reply blocks/large-manual-promo-block/features/large-manual-promo/default.jsx]: Removed it.

File: blocks/large-manual-promo-block/features/large-manual-promo/default.jsx (hunk 5 above)
Comment (malavikakoppula) [reply blocks/large-manual-promo-block/features/large-manual-promo/default.jsx]: Removed it.

File: blocks/large-manual-promo-block/features/large-manual-promo/default.jsx (hunk 6 above)
Comment (malavikakoppula) [reply blocks/large-manual-promo-block/features/large-manual-promo/default.jsx]: Removed it.

File: blocks/large-manual-promo-block/features/large-manual-promo/default.jsx (hunk 7 above)
Comment (malavikakoppula) [reply blocks/large-manual-promo-block/features/large-manual-promo/default.jsx]: Removed it.

File: blocks/large-manual-promo-block/features/large-manual-promo/default.jsx (hunk 8 above)
Comment (malavikakoppula) [reply blocks/large-manual-promo-block/features/large-manual-promo/default.jsx]: Removed it.

File: blocks/large-manual-promo-block/features/large-manual-promo/default.jsx (hunk 9 above)
Comment (malavikakoppula) [reply blocks/large-manual-promo-block/features/large-manual-promo/default.jsx]: Removed it.

File: blocks/large-manual-promo-block/features/large-manual-promo/default.jsx (hunk 10 above)
Comment (malavikakoppula) [reply blocks/large-manual-promo-block/features/large-manual-promo/default.jsx]: Removed it.  Had to keep `BLOCK_CLASS_NAME` for gap. Just removed `BLOCK_CLASS_NAME}__${imagePosition}`

File: blocks/large-manual-promo-block/_index.scss [hunk 11]
```diff
@@ -0,0 +1,11 @@
+@use "@wpmedia/arc-themes-components/scss";
+
+.b-large-manual-promo {
+	&__info {
+		@include scss.block-properties("large-manual-promo-info");
+		@include scss.block-components("large-manual-promo-info");
+	}
```
Comment (JackHowa) [inline blocks/large-manual-promo-block/_index.scss]:
```suggestion
```

is this being used anywhere? if not, like the previous classes that were removed, this can be removed as well

File: blocks/large-manual-promo-block/README.md [hunk 12]
```diff
@@ -2,10 +2,6 @@
 
 _Block containing a large manual promo component. Please provide a 1-2 sentence description of what the block is and what it does._
```
Comment (JackHowa) [inline blocks/large-manual-promo-block/README.md]: can you update this to not include placeholder info. Do you remember what we talked about was placeholder info? if you need help writing this, please ask a dev or a product person

**Review by JackHowa — CHANGES_REQUESTED** [review]
Looks like a few more things need to be addressed based on the checklist do you remember discussion about documentation and placeholder content

File: blocks/large-manual-promo-block/README.md (hunk 12 above)
Comment (malavikakoppula) [reply blocks/large-manual-promo-block/README.md]: Updated README.

File: blocks/large-manual-promo-block/_index.scss (hunk 11 above)
Comment (malavikakoppula) [reply blocks/large-manual-promo-block/_index.scss]: oh I missed it.

File: blocks/large-manual-promo-block/features/large-manual-promo/default.jsx [hunk 13]
```diff
@@ -1,19 +1,96 @@
...
+	const PromoDescription = () => (showDescription ? <Paragraph>{description}</Paragraph> : null);
+
 	return (
-		<LazyLoad enabled={customFields?.lazyLoad && !isAdmin}>
-			<LargePromoPresentation {...customFields} />
+		<LazyLoad enabled={shouldLazyLoad}>
+			<HeadingSection>
+				<Grid role="article" className={BLOCK_CLASS_NAME}>
+					<PromoImage />
+					<Stack className={`${BLOCK_CLASS_NAME}__info`}>
```
Comment (JackHowa) [inline blocks/large-manual-promo-block/features/large-manual-promo/default.jsx]: like we've said, if the classname is not used, it can be removed

**Review by JackHowa — CHANGES_REQUESTED** [review]
Remember, if the classname is not used, it can be removed

File: blocks/large-manual-promo-block/_index.scss (hunk 11 above)
Comment (JackHowa) [reply blocks/large-manual-promo-block/_index.scss]: I updated as well and removed it from the jsx

File: blocks/large-manual-promo-block/features/large-manual-promo/default.jsx (hunk 13 above)
Comment (JackHowa) [reply blocks/large-manual-promo-block/features/large-manual-promo/default.jsx]: I fixed

Comment (malavikakoppula) [reply blocks/large-manual-promo-block/features/large-manual-promo/default.jsx]: Thank you For fixing it

**Comment by JackHowa** [comment]
@malavikakoppula is working on matching the font-weight, line-height, and font-size of the overline, heading, and paragraph for mobile and desktop :)

**Comment by JackHowa** [comment]
@haganka can you and @malavikakoppula work together on this for design qa?

**Review by JackHowa — APPROVED** [review]
Great work! ready to go pending dqa approval

**Comment by JackHowa** [comment]
ready for design qa @haganka

**Review by JackHowa — CHANGES_REQUESTED** [review]
Adding the margin will always add the margin even when there's no description below the heading. can you think of another way of adding that space only when there's a description? There's still a way we can use gap as far as I understand but the structure may need to be different 

Screen Shot 2022-07-05 at 16 33 37

File: blocks/large-manual-promo-block/_index.scss [hunk 14]
```diff
@@ -0,0 +1,10 @@
+@use "@wpmedia/arc-themes-components/scss";
+
+.b-large-manual-promo {
+	@include scss.block-components("large-manual-promo");
+	@include scss.block-properties("large-manual-promo");
+}
+.b-large-manual-promo__text {
```
[the remark this answers is not in the document]
Comment (malavikakoppula) [reply blocks/large-manual-promo-block/_index.scss:8]: added new line

**Comment by malavikakoppula** [comment]
> Adding the margin will always add the margin even when there's no description below the heading. can you think of another way of adding that space only when there's a description? There's still a way we can use gap as far as I understand but the structure may need to be different
> 
> Screen Shot 2022-07-05 at 16 33 37

Fixed it by making right content as grid and applied gap with stack for both headline and description

**Review by JackHowa — APPROVED** [review]
Looks good if dqa approves!

File: .storybook/themes/news.scss [hunk 15]
```diff
@@ -879,6 +879,30 @@
 					font-size: var(--global-font-size-13),
 					font-weight: var(--global-font-weight-7),
 				),
+				large-manual-promo: (
```
Comment (JackHowa) [inline .storybook/themes/news.scss:882]: all of this should also be in feature pack

**Comment by JackHowa** [comment]
Good work @malavikakoppula ! make sure all of the styles here are in the feature pack -- ready to go!

File: .storybook/themes/news.scss (hunk 15 above)
Comment (malavikakoppula) [reply .storybook/themes/news.scss:882]: updated in the feature-pack
